aten/src/ATen/native/TensorShape.cpp

```diff
@@ -497,10 +497,10 @@ static Tensor cat_sparse_impl(TensorList tensors, int64_t dim) {
         idxs,
         vals,
         sizes_copy,
-        optTypeMetaToScalarType(tensors[0].options().dtype_opt()),
-        tensors[0].options().layout_opt(),
-        tensors[0].options().device_opt(),
-        tensors[0].options().pinned_memory_opt());
+        optTypeMetaToScalarType(tensors[0].get().options().dtype_opt()),
```

Comment:
The need to manually decay the reference wrapper here is kind of a drag :(

Reply:
Another way around this (still decaying, though) is to have a `const Tensor& first = tensors[0];`, and use `first` instead of `tensors[0].get()`.